File: jbake-core/src/test/java/org/jbake/TestUtils.java
package org.jbake;

import org.apache.commons.vfs2.util.Os;

import java.io.File;
import java.io.FilenameFilter;
import java.io.IOException;

public class TestUtils {

    /**
     * Hides the assets on Windows that start with a dot (e.g. .test.txt but not test.txt) so File.isHidden() returns true for those files.
     */
    public static void hideAssets(File assets) throws IOException, InterruptedException {
        if (isWindows()) {
            final File[] hiddenFiles = assets.listFiles(new FilenameFilter() {
                @Override
                public boolean accept(File dir, String name) {
                    return name.startsWith(".");
                }
            });
            for (File file : hiddenFiles) {
                final Process process = Runtime.getRuntime().exec(new String[]{"attrib", "+h", file.getAbsolutePath()});
                process.waitFor();
            }
        }
    }

    public static boolean isWindows() {
        return Os.isFamily(Os.OS_FAMILY_WINDOWS);
    }

    public static File getTestResourcesAsSourceFolder() {
        return getTestResourcesAsSourceFolder("/fixture");
    }

    public static File getTestResourcesAsSourceFolder(String name) {
        return new File(TestUtils.class.getResource(name).getFile());
    }

    public static File newFolder(File base, String folderName) {
        File templateFolder = new File(base,folderName);
        templateFolder.mkdir();
        return templateFolder;
    }


}

File: jbake-core/src/test/java/org/jbake/TestUtils.kt
package org.jbake

import org.apache.commons.lang3.SystemUtils
import java.io.File
import java.io.IOException
import java.nio.file.Path

object TestUtils {

    /**
     * Hides the assets on Windows that start with a dot (e.g. .test.txt but not test.txt) so File.isHidden() returns true for those files.
     */
    @Throws(IOException::class, InterruptedException::class)
    fun hideAssets(assets: File) {
        if (!isWindows) return
        val hiddenFiles = assets.listFiles { dir, name -> name.startsWith(".") }
        for (file in hiddenFiles) {
            Runtime.getRuntime().exec(arrayOf("attrib", "+h", file.absolutePath)).waitFor()
        }
    }

    val isWindows: Boolean
        get() = SystemUtils.IS_OS_WINDOWS

    val testResourcesAsSourceDir: File
        get() = getTestResourcesAsSourceDir("/fixture")

    fun getTestResourcesAsSourceDir(name: String): File {
        val resource = TestUtils::class.java.getResource(name) ?: throw IllegalArgumentException("Resource not found: $name")
        return File(resource.file)
    }

    fun newDir(base: File?, folderName: String): File {
        val templateDir = File(base, folderName)
        templateDir.mkdir()
        return templateDir
    }

    fun escapeBackSlashes(path: Path)
        = path.toString().let {
            if (!isWindows) it else it.replace("""\""", """\\""")
        }
}
